deals-add.js

```diff
@@ -0,0 +1,223 @@
+var script = document.createElement('script');
+script.type = 'text/javascript';
+script.src = 'https://s.cdpn.io/3/clamp.js';
+document.head.appendChild(script);
+
+var script = document.createElement('script');
+script.type = 'text/javascript';
+script.src = 'https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.20.1/moment-with-locales.js';
+document.head.appendChild(script);
+
+var dealsContainer = document.getElementsByClassName("deals-container")[0];
+var currentCategory = 'Daily';
+var offset = 0;
+var hasMore = true;
+var fetching = false;
+
+//Buttons
+document.getElementsByClassName("daily-button")[0].addEventListener("click", function() {
+	// reset feed to remove all cards currently on page
+	resetFeed();
+	currentCategory = 'Daily'
+	getDeals(currentCategory);
+
+});
+
+document.getElementsByClassName("price-drop-button")[0].addEventListener("click", function() {
+	resetFeed();
+	currentCategory = 'PriceDrop'
+	getDeals(currentCategory);
+});
+
+document.getElementsByClassName("popular-button")[0].addEventListener("click", function() {
+	resetFeed();
+	currentCategory = 'Popular'
+	getDeals(currentCategory);
+});
+
+// creates a new product card
+function createNewCard() {
+	var newCard = dealsContainer.firstElementChild.cloneNode(true);
+	dealsContainer.appendChild(newCard);
+}
+
+// displays the correct currency format
+function getCurrency(deal) {
+	if (deal.listing.currency == "SGD") {
+		return "S$";
+	}
+}
+
+// rounds the currency to the specified decimal places
+function round(value, decimals) {
+  var savings = Number(Math.round(value+'e'+decimals)+'e-'+decimals);
+  if (savings % 1.00 > 0) { // if there are decimals, display with 2 decimal places
+		return savings.toFixed(2);
+	}
+	else { // if there are no decimals, do not display decimal places
+		return savings;
+	}
+}
+
+// calculate amount of savings from previousPrice and currentPrice
```

Comment:
For pure functions such as these (i.e. given an input, there's always an output), try to write jsdoc type comments.
i.e.
```
  /**
   * Calculate savings for the deal based on current and previous price
   *
   * @param {Object}    deal         Object representation of the deal to calculate savings for
   * @returns {Number} Savings, rounded to 2 d.p
   */
```